services/brig/test/integration/API/User/Account.hs

```diff
@@ -1632,18 +1633,91 @@ testTooManyMembersForLegalhold opts brig = do
         const 403 === statusCode
         const (Right "too-many-members-for-legalhold") === fmap Wai.label . responseJsonEither
 
+testDeleteUserWithCompletelyDeletedUser :: Brig -> Cannon -> AWS.Env -> Http ()
+testDeleteUserWithCompletelyDeletedUser brig cannon aws = do
+  u <- randomUser brig
+  liftIO $ Util.assertUserJournalQueue "user activate testDeleteUserWithCompletelyDeletedUser" aws (userActivateJournaled u)
+  setHandleAndDeleteUser brig cannon u [] aws $
+    \uid -> deleteUserInternal uid brig !!! const 202 === statusCode
+  do
+    let uid = userId u
+    deleteUserInternal uid brig
+      !!! do
+        const 200 === statusCode
+
+testDeleteUserWithNoUser :: Brig -> Http ()
+testDeleteUserWithNoUser brig = do
+  nonExistingUid :: UserId <- liftIO $ generate arbitrary
+  deleteUserInternal nonExistingUid brig
+    !!! do
+      const 404 === statusCode
+
+testDeleteUserWithNotDeletedUser :: HasCallStack => Brig -> Cannon -> AWS.Env -> Http ()
+testDeleteUserWithNotDeletedUser brig cannon aws = do
+  u <- randomUser brig
+  liftIO $ Util.assertUserJournalQueue "user activate testDeleteUserWithNotDeletedUser" aws (userActivateJournaled u)
+  do
+    setHandleAndDeleteUser brig cannon u [] aws $
+      ( \uid' ->
+          deleteUserInternal uid' brig
+            !!! do
+              const 202 === statusCode
+      )
+
+testDeleteUserWithDanglingProperty :: Brig -> Cannon -> AWS.Env -> Http ()
+testDeleteUserWithDanglingProperty brig cannon aws = do
+  u <- randomUser brig
+  liftIO $ Util.assertUserJournalQueue "user activate testDeleteUserWithDanglingProperty" aws (userActivateJournaled u)
+
+  let uid = userId u
+  -- First set a unique handle (to verify freeing of the handle)
+  hdl <- randomHandle
+  let update = RequestBodyLBS . encode $ HandleUpdate hdl
+  put (brig . path "/self/handle" . contentJson . zUser uid . zConn "c" . body update)
+    !!! const 200 === statusCode
+
+  deleteUserInternal uid brig !!! const 202 === statusCode
+  liftIO $ Util.assertUserJournalQueue "user deletion testDeleteUserWithDanglingProperty" aws (userDeleteJournaled uid)
+
+  setProperty brig (userId u) "foo" objectProp
```

Comment:
interesting, these work with invalid `UserId`s?  ok, good for writing this test i guess...